test/hotspot/jtreg/runtime/Monitor/MonitorUsedDeflationThresholdTest.java

```diff
@@ -0,0 +1,138 @@
+            } catch (InterruptedException ie) {
+                // ignore InterruptedException
+            }
+            System.out.println("Done delaying for " + DELAY_SECS + " secs.");
+            return;
+        }
+
+        Object obj = new Object();
+
+        synchronized(obj) {
+            try {
+                obj.wait(1);  // force inflation
+            } catch (InterruptedException ie) {
+                // ignore InterruptedException
+            }
+            do_work(count + 1);
+        }
+    }
+
+    public static void usage() {
+        System.err.println("Usage: java " +
+                           "MonitorUsedDeflationThresholdTest inflate_count");
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            // Without args we invoke the test in a java sub-process:
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                // Test doesn't need much Java heap:
+                "-Xmx100M",
+                // AvgMonitorsPerThreadEstimate == 1 means we'll start with
+                // an in_use_list_ceiling of <n-threads> plus a couple of
+                // of monitors for threads that call Object.wait().
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:AvgMonitorsPerThreadEstimate=1",
+                // Enable monitorinflation logging so we can see that
+                // MonitorUsedDeflationThreshold and
+                // NoAsyncDeflationProgressMaxoption are working.
+                "-Xlog:monitorinflation=info",
+                // Enable some safepoint logging for diagnostic purposes.
+                "-Xlog:safepoint+cleanup=info",
+                "-Xlog:safepoint+stats=debug",
+                // Run the test with inflate_count == 33 since that
+                // reproduced the bug with JDK13. Anything above the
+                // in_use_list_ceiling will do the trick.
+                "MonitorUsedDeflationThresholdTest", "33");
+
+            OutputAnalyzer output_detail = new OutputAnalyzer(pb.start());
+
+            // This mesg means:
+            // - AvgMonitorsPerThreadEstimate == 1 reduced in_use_list_ceiling
+            //   to a small number.
+            // - and we crossed MonitorUsedDeflationThreshold:
+            output_detail.shouldMatch("begin deflating: .*");
+            System.out.println("Found beginning of a deflation cycle.");
+
+            // This mesg means we hit NoAsyncDeflationProgressMax and
+            // had to adjust the in_use_list_ceiling:
+            String too_many = output_detail.firstMatch("Too many deflations without progress; .*", 0);
+            if (too_many == null) {
+                throw new RuntimeException("Did not find too_many string in output.\n");
+            }
+            System.out.println("too_many='" + too_many + "'");
+
+            System.out.println("PASSED.");
+            return;
+        }
+
```

Comment:
Please add a comment e.g.
`// else we are the exec'd java subprocess, so run the actual test

Reply:
Added.